app/routes.py

```diff
@@ -1,2 +1,258 @@
+
+    elif request.method == "GET":
+        return video.make_json()
+    
+    elif request.method == "PUT":
+        request_body = request.get_json()
+
+        if "title" not in request_body.keys() \
+            or "release_date" not in request_body.keys() \
+                or "total_inventory" not in request_body.keys():
+                    # or request_body["video_id"] is not int:
+                return make_response({"details": "Invalid data"}, 400) 
+        
+        video.title = request_body["title"]
+        video.release_date = request_body["release_date"]
+        video.total_copies = request_body["total_inventory"]
+        return video.make_json()
+    
+    elif request.method == "DELETE":
+        db.session.delete(video)
+        db.session.commit()
+        return video.return_id()
+
+@videos_bp.route("/<id>/rentals", methods = ["GET"])
+def handle_rentals_by_video(id):
+    """
+    takes in video id
+    returns a list of customers that have the video
+    at the given id checked-out
+    """
+    video = Video.query.get(id)
+    
+    if video is None:
+        return make_response("", 404)
+
+    customers_renting = video.customers
+
+    customer_list = []
+    for rental in customers_renting:
+        customer_list.append({
+            "due_date": rental.due_date, 
+            "name": rental.customer.name,
+            "phone": rental.customer.phone_number,
+            "postal_code": rental.customer.postal_code
+        })
+
+    return jsonify(customer_list)
+
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+
+@rentals_bp.route("/check-out", methods = ["POST"])
+def rental_check_out():
+    """
+    Check's out one video to one customer
+    """
+    request_body = request.get_json()
+    try:
+        customer_id = int(request_body["customer_id"])
+        video_id = int(request_body["video_id"])
+    except ValueError:
+        return make_response({"details": "Invalid data"}, 400)
+
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    if customer is None or video is None:
+        return make_response("", 404)
+
+    elif video.get_available_inventory() == 0:
+        return make_response({"details":"No copies of this video are available"}, 400)
+        
+    else:
+        customer.check_out()
+        new_rental= Rental(customer_id=customer_id, video_id=video_id, due_date=(datetime.now() + timedelta(days=7)))
+
+        db.session.add(new_rental)
+        db.session.commit()
+
+        return make_response(new_rental.make_json())
+
+@rentals_bp.route("/check-in", methods = ["POST"])
+def rental_check_in():
+    """
+    checks in one video to one customer
+    """
+    request_body = request.get_json()
+
+    try:
+        customer_id = int(request_body["customer_id"])
+        video_id = int(request_body["video_id"])
+    except ValueError or KeyError:
+        return make_response({"details": "Invalid data"}, 400)
```

Comment:
Consider handling ValueErrors and KeyError separately to provide more details about the invalid data.